### PR TITLE
New teams

### PR DESCRIPTION
**Pull request by ryjones**
Creating teams for repos

### CODE REVIEW

**Comment by ryjones** [comment]
@baentsch @bhess this is to move people from collaborator roles to teams with roles. The level of access should remain the same - most of the collaborators were already members of a team with identical access.

**Comment by ryjones** [comment]
@baentsch please review

**Review by baentsch — COMMENTED** [review]
As far as oqsprovider is concerned, this seems to go from too few permissions for too few people to too many permissions for too many people. Is there no way to model exactly what's documented in GOVERNANCE.md (unless answers to #45 contradict that)?

**Review by dstebila — APPROVED** [review]

File: config.yaml [hunk 1]
```diff
@@ -118,11 +140,6 @@ repositories:
     tsc: read
   visibility: public
 - name: liboqs
-  collaborators:
-    Martyrshot: write
```
Comment (dstebila) [inline config.yaml:122]: @Martyrshot seems to be deleted as a collaborator but not added to a team.

**Review by dstebila — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 1 above)
Comment (ryjones) [reply config.yaml:122]: [He is already a member](https://github.com/open-quantum-safe/tsc/blob/6dc9cd8c6c443e005f0526e5215f054c3554c7f1/config.yaml#L21) of l[ib-oqs-committers](https://github.com/open-quantum-safe/tsc/blob/6dc9cd8c6c443e005f0526e5215f054c3554c7f1/config.yaml#L136) which maintains his access.

Comment (dstebila) [reply config.yaml:122]: Whoops, I missed that. Sorry.

**Review by dstebila — APPROVED** [review]

File: config.yaml (hunk 1 above)
Comment (Martyrshot) [reply config.yaml:122]: Thanks for looking out for me @dstebila 😄

**Comment by baentsch** [comment]
@dstebila Can you please explain why you think it's a good idea to give admin rights to oqs-provider to everyone with a "code-owner" designation? If I'm not wrong, you now approved all contributors to have even more privileges than normal GH maintainer(s): This is IMO the exact opposite of the least privilege concept and also a direct contradiction to [GOVERNANCE.md](https://github.com/open-quantum-safe/oqs-provider/blob/main/GOVERNANCE.md). The latter irks me even more than the security downgrade: Why did I write this in the first place if not to provide some "guard rails" protecting the code?

**Comment by dstebila** [comment]
> @dstebila Can you please explain why you think it's a good idea to give admin rights to oqs-provider to everyone with a "code-owner" designation? If I'm not wrong, you now approved all contributors to have even more privileges than normal GH maintainer(s): This is IMO the exact opposite of the least privilege concept and also a direct contradiction to [GOVERNANCE.md](https://github.com/open-quantum-safe/oqs-provider/blob/main/GOVERNANCE.md). The latter irks me even more than the security downgrade: Why did I write this in the first place if not to provide some "guard rails" protecting the code?

You're right. What level of access should the codeowners team have?

**Comment by baentsch** [comment]
> You're right. What level of access should the codeowners team have?

Codeowners in my eyes are Contributors and should have the same rights. But I already don't understand the separation of "codeowners" and "contributors" to begin with. This only exists in oqsprovider, no-where else: Why?

In general, I think the whole file is a big mess, completely un-structured, not orthogonal across projects and not in line with GOVERNANCE.md files where present. Yes, it did become better recently with the introduction of teams (though not consistently, see e.g., boringssl), but still, I don't understand why there's not a simple, orthogonal/per-project "maintainers", "committers", and "triage" category with admin (or maintain, if no sufficient trust exists), write, and triage rights, respectively as per our governance rules? Related question: Why is there a need to sometime have "admin" and sometime "maintainer" and sometime both teams per sub-project? Then: Why is there absolutely no documentation (e.g., as to exceptions to the "orthogonality" rule)?
This "orthogonality" can be simplified by having a single triage team triaging all OQS projects (as is the case if I get it right) and a single language-wrapper sub-project "designator" for all language-wrapper projects but rust (not the case right now, if I get it right). These changes alone should cut the size of the file substantially and make it understandable to the unwary (or at least me :). In any case, it should _never_ violate GOVERNANCE.md files as it did from the very start until now. Lastly, I don't understand why an admin often (but not always) is maintainer (e.g., of teams): If this should be managed by the community, only community members (arguably maintainers) should be listed. I understand it was started by an admin to manage the multitude of projects, but at no time did I see agreement (and adherence) to basic principles like these:

- Never violate GOVERNANCE.md files' contents where present
- Keep things orthogonal across projects -- if not, document by comment why it's not possible so everyone can understand exceptions
- Do not have LF admins overrule maintainer's rights

OK, the latter is probably just my problem since the don't-trust-@baentsch principle started in April when all my permissions were withdrawn, so probably unrealistic/unacceptable to LF. Nevertheless I think it would be a logical principle _if_ the community and not LF admins should control the project('s access rights). Conceptually, what about an OQS-maintainers team listed as maintainer for all sub-project's maintainer team lists? In my "ideal" world, every person would be listed in as few teams as possible, ideally just one -- every additional use of a specific person's ID should be accompanied by a comment. This way, it becomes obvious when specific persons become "risky" (in terms of "key person risk") and the community aware of the need to do something about it.

As usual, sorry for the long text & disregard as you see fit.

**Comment by SWilson4** [comment]
Perhaps our Committers ought in fact to have "maintain" privileges, Maintainers ought to have "admin" privileges, and Contributors (incl. people in the codeowners file) ought to have "write" privileges? (See [here](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role) for documentation on perms.)

**Pull request merged**